### DISCUSSION TITLE
Mirth Connect testing automation

### DISCUSSION THREAD

**Discussion by muralikunda** [nextgenhealthcare/connect#4940] [Q&A]
Hi 
I am new to Mirth and I am coming with heavy UI automation background using Selenium like tools. Is there any UI automation framework or tools to automate Mirth testing in different channels.

Appreciate any input.

Thanks

**Comment by tom08zehn** [comment]
No. I was looking for something similar quite a long time and tried all variants but none of them can do the job. The problem is that you cannot run external code in the "Mirth context"... also because there's not just one context but many (global, channel, message, etc.).

I'm using a helper channel that accepts a channel name (what channel to test) + configuration variables (changes the channel's behavior) + message. This is exposed as a kind of HTTP REST API to external test frameworks.

The helper channel forwards the message to the destination channel and returns the response. Also, I introduced kind of "invisible break points" into the channels that can be tested. These "break points" are writing debug information into the channel's `channelMap`. The helper channel provides the value of this "debug log" to the external framework. The test framework can follow what the destination channel did and if error occured.
